ckanext/dcatapchharvest/profiles.py

```diff
@@ -357,6 +357,10 @@ def parse_dataset(self, dataset_dict, dataset_ref):  # noqa
 
         # Relations
         dataset_dict['relations'] = self._relations(dataset_ref, DCT.relation)
+        for i in range(len(dataset_dict['relations'])):
+            if dataset_dict['relations'][i-1]['label'] == {}:
+                dataset_dict['relations'][i-1]['label'] = \
+                    str(dataset_dict['relations'][i-1].get('url'))
```

Comment:
The logic here is OK, but this seems like a slightly complicated way to do this. Why not avoid the `for i in range` stuff and do something like this?

```
for relation in dataset_dict['relations]:
    if relation['label'] == {}:
        relation['label'] = str(relation.get('url', '')
```

I added the `''` to the `relation.get('url')` because the default return value is `None`. Just in case we ever get a relation that doesn't have a url value, this is a bit safer.

Last thing - `range(n)` returns values starting at 0 and ending at `n-1`, so using `[i-1]` in your code above is unnecessary. You can just use `[i]`. :)

Reply:
Make sense, thank you very much for this advice :)